app/helpers/main_site_helper.rb

```diff
@@ -36,6 +36,26 @@ def user_profile_link( user = current_user )
     link_to user_display_name( user ), user_profile_path( user.username )
   end
 
+  # Returns the path to a comment's parent resource, anchored to the comment
+  # Tries to fall back gracefully if comment is deleted/marked as spam
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def comment_in_context_path( comment )
```

Comment:
The responsibility for knowing how to generate a comment's URL belongs to the comment itself more than anything else. For this reason, consider moving some/all this URL generation logic into a method on the `Comment` model. This would also help with reuse in the specs mentioned in the TODO comment below.